nixos/tests/ayatana-indicators.nix

```diff
@@ -0,0 +1,71 @@
+import ./make-test-python.nix ({ pkgs, lib, ... }: let
+  user = "alice";
+in {
+  name = "ayatana-indicators";
+
+  meta = {
+    maintainers = with lib.maintainers; [ OPNA2608 ];
+  };
+
+  nodes.machine = { config, ... }: {
+    imports = [
+      ./common/auto.nix
+      ./common/user-account.nix
+    ];
+
+    test-support.displayManager.auto = {
+      enable = true;
+      inherit user;
+    };
+
+    services.xserver = {
+      enable = true;
+      desktopManager.mate.enable = true;
+      displayManager.defaultSession = lib.mkForce "mate";
+    };
+
+    services.ayatana-indicators = {
+      enable = true;
+      packages = with pkgs; [
+        ayatana-indicator-messages
+      ];
+    };
+
+    # Services needed by some indicators
+    services.accounts-daemon.enable = true; # messages
+  };
+
+  # TODO session indicator starts up in a semi-broken state, but works fine after a restart. maybe being started before graphical session is truly up & ready?
+  testScript = { nodes, ... }: let
+    runCommandPerIndicatorService = command: lib.strings.concatMapStringsSep "\n" command nodes.machine.systemd.user.targets."ayatana-indicators".wants;
+  in ''
+    start_all()
+    machine.wait_for_x()
+
+    # Desktop environment should reach graphical-session.target
+    machine.wait_for_unit("graphical-session.target", "${user}")
+
+    # MATE relies on XDG autostart to bring up the indicators.
+    # Not sure *when* XDG autostart fires them up, and awaiting pgrep success seems to misbehave?
+    machine.sleep(10)
```

Comment:
flaky tests are very annoying on hydra, and this one is very much affected by the current workload of the host. Is it possible to await the mate user shell or a dummy XDG autostart target?

Reply:
Maybe? I'll have to look. Perhaps `xdg-desktop-autostart.target`, but I'm not sure MATE uses SystemD for XDG autostart. Not sure if there's anything for MATE itself.

Reply:
```suggestion
```

I think `wait_untill_succeeds` below is better

Reply:
We could instead `wait_until_succeeds("pgrep [...]")` below, and I would *like* the idea of not relying on `sleep()`s at all, but in my testing not `sleep()`ing anywhere will blow past *all* starts-but-immediately-segfaults situations. Even checking `coredumpctl` will report no errors, *unless* we `sleep()` to let it catch up with handling the coredumps.

Reply:
Great exploration of the problem space. Even forcing coredumpctl to have a look wouldn't work unless we can guarantee a deficit would have caused a coredump by then. `rg '\.sleep\(.*\)' -o | cut -d: -f2 | sort | uniq -c` reveals to me that 10 seconds is a pretty common wait time, so lets stick with it.